packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -723,17 +799,55 @@ class MemoryBodyState extends State<MemoryBody>
           const PaddedDivider(
             padding: EdgeInsets.only(left: denseRowSpacing),
           ),
-          Image(image: AssetImage(image)),
+          widget,
         ]);
       }
 
       return rightSide;
     }
 
+    Widget legendSymbol(Map<String, Object> dataToDisplay) {
+      final image = dataToDisplay.containsKey(renderImage)
+          ? dataToDisplay[renderImage] as String
+          : null;
+      final color = dataToDisplay.containsKey(renderLine)
+          ? dataToDisplay[renderLine] as Color
+          : null;
+      final dashedLine = dataToDisplay.containsKey(renderDashed)
+          ? dataToDisplay[renderDashed]
+          : false;
+
+      Widget traceColor;
+      if (color != null) {
+        if (dashedLine) {
+          traceColor = Padding(
+            child: CustomPaint(
+              painter: DashedLine(15, color, 2, 4, 3),
+              foregroundPainter: DashedLine(15, color, 2, 4, 3),
+            ),
+            padding: const EdgeInsets.fromLTRB(2, 5, 20, 1),
+          );
+        } else {
+          traceColor = Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 1.0),
+            child: Container(height: 6, width: 20, color: color),
```

Comment:
trailing comma always when defining widgets

Reply:
Done. I thought if the widget fits on a line we didn't add the trailing comma.